drivers/flash/soc_flash_nrf_ticker.c

```diff
@@ -0,0 +1,224 @@
+}
+
+static void time_slot_callback_work(uint32_t ticks_at_expire,
+				    uint32_t remainder,
+				    uint16_t lazy, void *context)
+{
+	struct flash_op_desc *op_desc;
+	uint8_t instance_index;
+	uint8_t ticker_id;
+
+	__ASSERT(ll_radio_state_is_idle(),
+		 "Radio is on during flash operation.\n");
+
+	op_desc = context;
+	if (op_desc->handler(op_desc->context) == FLASH_OP_DONE) {
+		ll_timeslice_ticker_id_get(&instance_index, &ticker_id);
+
+		/* Stop the time slot ticker */
+		_ticker_stop(instance_index, 0, ticker_id);
+
+		_ticker_synch_context.result = 0;
+
+		/* notify thread that data is available */
+		k_sem_give(&sem_sync);
+	}
+}
+
+static void time_slot_delay(uint32_t ticks_at_expire, uint32_t ticks_delay,
+			    ticker_timeout_func callback, void *context)
+{
+	uint8_t instance_index;
+	uint8_t ticker_id;
+	int err;
+
+	ll_timeslice_ticker_id_get(&instance_index, &ticker_id);
+
+	/* start a secondary one-shot ticker after ticks_delay,
+	 * this will let any radio role to gracefully abort and release the
+	 * Radio h/w.
+	 */
+	err = ticker_start(instance_index, /* Radio instance ticker */
+			   0, /* user_id */
+			   (ticker_id + 1), /* ticker_id */
+			   ticks_at_expire, /* current tick */
+			   ticks_delay, /* one-shot delayed timeout */
+			   0, /* periodic timeout  */
+			   0, /* periodic remainder */
+			   0, /* lazy, voluntary skips */
+			   0,
+			   callback, /* handler for executing radio abort or */
+				     /* flash work */
+			   context, /* the context for the flash operation */
+			   NULL, /* no op callback */
+			   NULL);
+
+	if (err != TICKER_STATUS_SUCCESS && err != TICKER_STATUS_BUSY) {
+		_ticker_synch_context.result = 0;
+
+		/* abort flash timeslots */
+		_ticker_stop(instance_index, 0, ticker_id);
+
+		/* notify thread that data is available */
+		k_sem_give(&sem_sync);
+	}
+}
+
+static void time_slot_callback_abort(uint32_t ticks_at_expire,
+				     uint32_t remainder,
+				     uint16_t lazy, void *context)
+{
+	ll_radio_state_abort();
+	time_slot_delay(ticks_at_expire,
+			HAL_TICKER_US_TO_TICKS(FLASH_RADIO_WORK_DELAY_US),
+			time_slot_callback_work,
+			context);
+}
+
+static void time_slot_callback_prepare(uint32_t ticks_at_expire,
+				       uint32_t remainder,
+				       uint16_t lazy, void *context)
+{
+#if defined(CONFIG_BT_CTLR_LOW_LAT)
+	time_slot_callback_abort(ticks_at_expire, remainder, lazy, context);
+#else /* !CONFIG_BT_CTLR_LOW_LAT */
+	time_slot_delay(ticks_at_expire,
+			HAL_TICKER_US_TO_TICKS(FLASH_RADIO_ABORT_DELAY_US),
+			time_slot_callback_abort,
+			context);
+#endif /* CONFIG_BT_CTLR_LOW_LAT */
+}
+
+
+int nrfsb_synch_backend_init(void)
+{
+	k_sem_init(&sem_sync, 0, 1);
+	return 0;
+}
+
+void nrfsb_set_synch_context(uint32_t duration)
+{
+
+	/* FLASH_SYNCH_SWITCHING_TIME is delay which is allways added by
+	 * the slot calling mechanism
+	 */
+	_ticker_synch_context.interval = duration - FLASH_SYNCH_SWITCHING_TIME;
+	_ticker_synch_context.slot = duration;
+}
```

Comment:
Maybe give these `inline` specifier, inlining this will take less space than processing call.